test/test_remote_io.py

```diff
@@ -223,24 +225,43 @@ def _filepath_fn(url):
             res = list(dl)
             self.assertEqual(sorted(expected), sorted(res))
 
+    def __get_s3_cnt(self, s3_pths: list):
```

Comment:
You might want to add an argument as `recursive`. IIRC, `fsspec` would behave like `recursive=False` but `S3FileLister` would behave like `recursive=True`.